richtext/plugins/org.polarsys.kitalpha.richtext.widget/src/org/polarsys/kitalpha/richtext/widget/internal/ListenerInstaller.java

```diff
@@ -54,6 +67,110 @@ public void createAllListeners(final MDENebulaBasedRichTextWidget widget) {
 		createSetDataEventListener(widget);
 	}
 
+	class OneTagChecker extends DefaultHandler {
+		private boolean onlyOneTag = true;
+
+		private boolean inRoot = false;
+
+		final private static String ROOT_ELEMENT = "clipboard";
+
+		private String expectedTag;
+
+		public OneTagChecker(String tag) {
+			expectedTag = tag;
+		}
+
+		@Override
+		public void startDocument() throws SAXException {
+			super.startDocument();
+			onlyOneTag = true;
+			inRoot = false;
+		}
+
+		@Override
+		public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
+			super.startElement(uri, localName, qName, attributes);
+			if (inRoot) {
+				if (!qName.equals(expectedTag)) {
+					onlyOneTag = false;
+				}
+			} else {
+				if (qName.equals(ROOT_ELEMENT)) {
+					inRoot = true;
+				}
+			}
+		}
+
+		@Override
+		public void endElement(String uri, String localName, String qName) throws SAXException {
+			if (qName.equals(ROOT_ELEMENT)) {
+				inRoot = false;
+			}
+			super.endElement(uri, localName, qName);
+		}
+
+		public boolean containOnlyOneTag() {
+			return onlyOneTag;
+		}
+	}
+
+	/**
+	 * This method check the content of the clipboard and determine whether it is really necessary to prompt the cleanup
+	 * dialog to the user. Currently this method avoid to display the prompt when the clipboard contains only links to
+	 * avoid this prompt when the clipboard contain links to Capella elements.
+	 * 
+	 * @param currentShouldCleanup
+	 *            the actual value of editor.config.pasteFromWordPromptCleanup
+	 * @return the new value to assign to editor.config.pasteFromWordPromptCleanup
+	 */
+	private boolean shouldCleanupClipboard(final boolean currentShouldCleanup) {
+		// If the cleanup has already been canceled (cleanup value is false),
+		// we can directly return false and avoid computing anything.
+		if (!currentShouldCleanup) {
+			return currentShouldCleanup;
+		} else {
+			// get html clipboard
+			HTMLTransfer htmlTransfer = HTMLTransfer.getInstance();
+			Clipboard clipboard = new Clipboard(Display.getCurrent());
+			String htmlData = (String) clipboard.getContents(htmlTransfer);
+			clipboard.dispose();
+
+			if (htmlData != null) {
+				// we put htmlData in clipboard to have root tag and to avoid parsing error.
+				String htmlDocument = "<clipboard>" + htmlData + "</clipboard>";
+				try {
+					// parse the clipboard to determine if there is only links inside
+					SAXParserFactory factory = SAXParserFactory.newInstance();
+					factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+					SAXParser saxParser = factory.newSAXParser();
+					OneTagChecker checker = new OneTagChecker("a");
```

Comment:
Can you fix the blocking sonar by adding 
see saxParser.setProperty like :
https://github.com/eclipse/kitalpha/blob/357c2828425ad73313b7bd8f70b0bc46c45222d9/detachment/plugins/org.polarsys.kitalpha.model.common.share/src/org/polarsys/kitalpha/model/common/share/modelresources/helpers/DetachmentHelper.java#L134-L135

Reply:
Ok, I pushed new commit, there shouldn't be any more sonar issues.